File: 算法/C/第5章 动态规划/5.3 路径问题/path.c
#include <stdio.h>

int uniquePath(int m, int n) {
    int f[m][n];
    for(int i = 0; i < m; i++) {
        for(int j = 0; j < n; j++) {
            if(i == 0 || j == 0) {
                f[i][j] = 1;
            } else {
                f[i][j] = f[i-1][j] + f[i][j-1];
            }
        }
    }
    return f[m-1][n-1];
}

int main() {
    printf("%d\n", uniquePath(4, 8));
    return 0;
}

File: old_version/Python/第5章 动态规划/5.3 路径问题/path.py
def unique_path(m, n):
    f = [[1] * n] + [[1] + [0] * (n - 1) for _ in range(m - 1)]
    for i in range(1, m):
        for j in range(1, n):
            f[i][j] = f[i-1][j] + f[i][j-1]
    return f[m-1][n-1]

def main():
    print(unique_path(4, 8))

if __name__ == "__main__":
    main()